src/integration_tests/mission_takeoff_land.cpp

```diff
@@ -0,0 +1,180 @@
+            const auto system = mavsdk.systems().at(0);
+
+            if (system->is_connected()) {
+                LogInfo() << "Discovered system";
+                prom->set_value();
+            }
+        });
+
+        ConnectionResult ret = mavsdk.add_udp_connection();
+        ASSERT_EQ(ret, ConnectionResult::Success);
+
+        auto status = future_result.wait_for(std::chrono::seconds(2));
+        ASSERT_EQ(status, std::future_status::ready);
+        future_result.get();
+        // FIXME: This hack is to prevent that the promise is set twice.
+        mavsdk.subscribe_on_new_system(nullptr);
+    }
+
+    auto system = mavsdk.systems().at(0);
+    auto telemetry = std::make_shared<Telemetry>(system);
+    auto mission = std::make_shared<Mission>(system);
+    auto action = std::make_shared<Action>(system);
+
+    PathChecker pc;
+
+    telemetry->subscribe_position([&pc](Telemetry::Position position) {
+        pc.check_current_alitude(position.relative_altitude_m);
+    });
+
+    std::vector<Telemetry::LandedState> landed_states_template = {
+        Telemetry::LandedState::OnGround,
+        Telemetry::LandedState::TakingOff,
+        Telemetry::LandedState::InAir,
+        Telemetry::LandedState::Landing,
+        Telemetry::LandedState::OnGround,
+        Telemetry::LandedState::TakingOff,
+        Telemetry::LandedState::InAir,
+        Telemetry::LandedState::Landing,
+        Telemetry::LandedState::OnGround};
+
+    std::vector<Telemetry::LandedState> landed_states;
+    Telemetry::LandedState landed_state_previous = Telemetry::LandedState::Unknown;
+    telemetry->subscribe_landed_state(
+        [&landed_states, &landed_state_previous](Telemetry::LandedState landed_state) {
+            if (landed_state_previous != landed_state) {
+                LogInfo() << landed_state;
+                landed_state_previous = landed_state;
+                landed_states.push_back(landed_state);
+            }
+        });
+
+    while (!telemetry->health_all_ok()) {
+        LogInfo() << "Waiting for system to be ready";
+        LogDebug() << "Health: " << telemetry->health();
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+
+    LogInfo() << "System ready";
+    LogInfo() << "Creating and uploading mission";
+
+    Mission::MissionItem new_item{};
+    Mission::MissionPlan mission_plan{};
+
+    new_item.latitude_deg = 47.398100;
+    new_item.longitude_deg = 8.545649;
+    new_item.relative_altitude_m = mission_altitude_m;
+    new_item.acceptance_radius_m = 1;
+    new_item.vehicle_action = Mission::MissionItem::VehicleAction::Takeoff;
+    mission_plan.mission_items.push_back(new_item);
+
+    new_item.latitude_deg = 47.398300;
+    new_item.vehicle_action = Mission::MissionItem::VehicleAction::Land;
+    mission_plan.mission_items.push_back(new_item);
+
+    new_item.latitude_deg = 47.398500;
+    new_item.vehicle_action = Mission::MissionItem::VehicleAction::TransitionToFw;
```

Comment:
I don't understand this action here. So we landed and now we're transitioning to fixedwing on ground? I don't think that's something PX4 supports. Usually, you need to takeoff and land in multicopter mode and then transition in air.

Also, if this test is made for a VTOL it needs to be named accordingly, like this:
https://github.com/mavlink/MAVSDK/blob/9b4b79cbdfca361fab9a528ba78e81376503cbd0/src/integration_tests/action_transition_multicopter_fixedwing.cpp#L10

Reply:
This is mistake. There should be Takeoff action. Will fix it.